### PR TITLE
Document the deprecated kubernetes.io/psp annotation

### PR DESCRIPTION
**Pull request by mtardy**
Solves #30578 according to that comment https://github.com/kubernetes/website/issues/30578#issuecomment-974691745.

I took the liberty to change a command in the tutorial as well because I added a step to check that annotation.
```diff
- kubectl-admin create -f example-psp.yaml
+ kubectl-admin create -f https://k8s.io/examples/policy/example-psp.yaml
```

### CODE REVIEW

File: content/en/docs/concepts/security/pod-security-policy.md [hunk 1]
```diff
@@ -214,6 +214,9 @@ controller selects policies according to the following criteria:
 2. If the pod must be defaulted or mutated, the first PodSecurityPolicy
    (ordered by name) to allow the pod is selected.
 
+When a Pod is validated against a PodSecurityPolicy, [a `kubernetes.io/psp` annotation](/docs/reference/labels-annotations-taints/#kubernetes-io-psp)
+is added with its name as its value.
```
Comment (sftim) [inline content/en/docs/concepts/security/pod-security-policy.md:218]:
```suggestion
is added to the Pod, with the name of the PodSecurityPolicy as the annotation value.
```

File: content/en/docs/concepts/security/pod-security-policy.md [hunk 2]
```diff
@@ -332,7 +335,15 @@ The output is similar to this
 pod "pause" created
 ```
 
-It works as expected! But any attempts to create a privileged pod should still
+It works as expected! You can verify that the pod was validated against the
+newly created PodSecurityPolicy:
+
+```shell
+kubectl-user get pod pause -o yaml | grep kubernetes.io/psp
+kubernetes.io/psp: example
```
Comment (sftim) [inline content/en/docs/concepts/security/pod-security-policy.md:343]: Please [separate the command from its output](https://kubernetes.io/docs/contribute/style/style-guide/#separate-commands-from-output).

File: content/en/docs/reference/labels-annotations-taints/_index.md [hunk 3]
```diff
@@ -618,6 +618,12 @@ or updating objects that contain Pod templates, such as Deployments, Jobs, State
 See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
 for more information.
 
+### kubernetes.io/psp (deprecated) {#kubernetes-io-psp}
+
+Example: `kubernetes.io/psp: restricted`
+
+Value is the name of the [PodSecurityPolicy](/docs/concepts/security/pod-security-policy/) that was validated against the ressource.
```
Comment (sftim) [inline content/en/docs/reference/labels-annotations-taints/_index.md:625]: I'd write:
```suggestion
This annotation is only relevant if you are using [PodSecurityPolicies](/docs/concepts/security/pod-security-policy/).

When the PodSecurityPolicy admission controller admits a Pod, the admission controller
modifies the Pod to have this annotation.
The value of the annotation is the name of the PodSecurityPolicy that was used for validation.
```

File: content/en/docs/concepts/security/pod-security-policy.md [hunk 4]
```diff
@@ -255,7 +258,7 @@ The name of a PodSecurityPolicy object must be a valid
 And create it with kubectl:
 
 ```shell
-kubectl-admin create -f example-psp.yaml
+kubectl-admin create -f https://k8s.io/examples/policy/example-psp.yaml
```
Comment (sftim) [inline content/en/docs/concepts/security/pod-security-policy.md:261]: You can skip “Define the example PodSecurityPolicy object in a file” too.

(nit: this tidying, to avoid having the reader paste in a manifest, would be even better if done in its own commit)

**Review by sftim — COMMENTED** [review]
Thanks!

/lgtm

I have a few suggestions if you want to tweak this further.

File: content/en/docs/concepts/security/pod-security-policy.md (hunk 2 above)
Comment (mtardy) [reply content/en/docs/concepts/security/pod-security-policy.md:343]: I did it also for the rest, I am not sure how much "the output is similar to this" I should imply instead of writing but this seems ok!

Comment (mtardy) [reply content/en/docs/concepts/security/pod-security-policy.md:365]: https://github.com/kubernetes/website/pull/34675/commits/8a4e62fb766e8efc296ca91823b210941fb24bcd

File: content/en/docs/concepts/security/pod-security-policy.md (hunk 4 above)
Comment (mtardy) [reply content/en/docs/concepts/security/pod-security-policy.md:260]: https://github.com/kubernetes/website/pull/34675/commits/1d55061a5a748bd9cd953470c847ac012cc2cf8f

**Review by divya-mohan0209 — COMMENTED** [review]
Thank you for the fix, Mahé!

/approve
